src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -274,6 +276,24 @@ struct context
         this->current_device = std::make_shared<hip_device>(0, n_streams);
     }
 
+    void wait_for(any_ptr queue)
+    {
+        auto status = hipEventRecord(begin_event.get(), queue.get<hipStream_t>());
+        if(status != hipSuccess)
+            MIGRAPHX_THROW("failed to record " + hip_error(status));
+
+        get_stream().wait(begin_event.get());
+    }
+
+    void finish_on(any_ptr queue)
+    {
+        get_stream().record(finish_event.get());
```

Comment:
Should this be 
```
auto status = hipEventRecord(begin_event.get(), queue.get<hipStream_t>());
```

Reply:
Yes but I'm using the record function already implemented for the current stream this is running on.

Reply:
Okay, I think i understood things other way around. I was thinking `queue` passed as parameter would be the one that will get used to run the computation. But it is the `get_queue()` one. 

So, in the `eval()` method in `program.cpp`.  Shouldn't it call `wait_for` and `finish_on` one after the another ? 

e.g. right now, it is doing `wait_for()`  that make sure the user-queue has finished all its commands. 

Then, program computes its output using `get_stream()` queue. 

and then, it calls `finish_on` on the user-queue, which makes sure that the user-queue doesn't run until `get_queue` has finished all its commands. but Is there anything that is blocking issues commands to the user-queue when the program is being run with `get_queue`?